raiden/network/transport/matrix/transport.py

```diff
@@ -529,14 +529,20 @@ def stop(self) -> None:
         # Wait for retriers which already started to exit, then discard them
         # Retriers which have not been started yet just get discarded
         # In the meanwhile no other retriers should be started since stop_event is set
-        gevent.wait({r.greenlet for r in self._address_to_retrier.values() if r.greenlet})
+        #
+        # No need to get on them, exceptions are re-raised because of the
```

Comment:
I don't know if this is the best approach, but I did not want to change how it is done, so I just added a comment as a heads up and maintained the code using the `link_exception`.